reward/src/lib.rs

```diff
@@ -0,0 +1,524 @@
+    }
+
+    if term::confirm("Do you wish to proceed?") {
+        let sum = amounts
+            .iter()
+            .fold(Some(U256::from(0_u64)), |sum, amt| {
+                sum.and_then(|s| s.checked_add(*amt))
+            })
+            .ok_or_else(|| anyhow!("Failed to sum all transactions amounts"))?;
+
+        let token: ERC20<SignerMiddleware<_, _>> = ERC20::new(signer.clone(), address);
+        let spinner = term::spinner(&term::format::tertiary("Checking allowance..."));
+        let allowance = rt.block_on(token.get_allowance(signer.address(), *SUPERSEEDER_ADDRESS))?;
+        spinner.finish();
+
+        if allowance < sum {
+            term::blank();
+            term::info!(
+                "{}",
+                term::format::tertiary("Approving transaction amount for Superseeder...")
+            );
+            term::blank();
+
+            let call = token.approve(*SUPERSEEDER_ADDRESS, sum)?;
+            rt.block_on(term::ethereum::transaction(call))?;
+        }
+
+        // generate tx(s)
+        let superseeder: Superseeder<SignerMiddleware<_, _>> = Superseeder::new(signer);
+        let call = superseeder.send(address, receivers, amounts)?;
+        term::blank();
+        term::info!("{}", term::format::tertiary("Sending transaction..."));
+        term::blank();
+        rt.block_on(term::ethereum::transaction(call))?;
+
+        // tag last processed commit
+        if let Some(head_sha1) = head_sha1 {
+            let spinner = term::spinner(&term::format::tertiary(
+                "Tagging the tip of range as `rad/reward`",
+            ));
+            let head_commit = repo.find_commit(head_sha1)?.into_object();
+            repo.tag_lightweight("rad/reward", &head_commit, true)?;
+            spinner.finish();
+        }
+    }
+
+    Ok(())
+}
+
+fn calculate_rewards(
+    strategy: Strategy,
+    reward: U256,
+    all_ens: &[&Ens],
+) -> anyhow::Result<HashMap<String, U256>> {
+    let mut shares: HashMap<String, usize> = HashMap::new();
+    let _ = all_ens
+        .iter()
+        .map(|ens| {
+            if let Some(v) = shares.get_mut(&ens.name) {
+                *v += 1;
+            } else {
+                shares.insert(ens.name.clone(), 1);
+            }
+        })
+        .collect::<Vec<()>>();
+
+    let mut rewards: HashMap<String, U256> = HashMap::new();
+    if strategy == Strategy::Equally {
+        let count = U256::from(shares.len());
+        let each_reward = reward
+            .checked_div(count)
+            .ok_or_else(|| anyhow!("Couldn't divide total reward by number of contributors"))?;
+
+        let _ = shares
+            .keys()
+            .map(|ens| {
+                rewards.insert(ens.to_string(), each_reward);
+            })
+            .collect::<Vec<()>>();
+    } else if strategy == Strategy::Weighted {
+        let total_shares = shares.values().sum::<usize>();
+        let reward_per_share = reward.checked_div(U256::from(total_shares)).unwrap();
```

Comment:
Why use a `checked_div` if we are unwrapping? The regular `div` will panic already

Reply:
https://github.com/radicle-dev/radicle-cli/commit/30805149493b3e5bd4eebcd5afb5746c9b987c3f#diff-26d7e6053383a7793e880ae8b6ee5d2f7d7c23ccd5e88e43f44052374e9f0f58R437-R439